packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -300,17 +301,23 @@ if (__DEV__) {
       );
       return;
     }
+    const type = fiber.type;
 
-    // Dedup strategy: Warn once per component.
-    if (didWarnAboutLegacyContext.has(fiber.type)) {
+    // Update legacy context counts
+    const warningCount = legacyContextCounts.get(type) || 0;
+    // Increase warning count by 1
+    legacyContextCounts.set(type, warningCount + 1);
+
+    // Dedup strategy: Warn once per component
+    if (didWarnAboutLegacyContext.has(type)) {
```

Comment:
Seems like we should do this check before incrementing the count.

Else we might end up including this type in the warning later, if another type has a context warning too?

Reply:
Good point. Thanks :)